expr/boolean.go

```diff
@@ -114,6 +114,46 @@ func CompareInt64(op string, pattern int64) (Boolean, error) {
 	}, nil
 }
 
+func CompareTime(op string, pattern int64) (Boolean, error) {
+	CompareInt, ok1 := compareInt[op]
+	CompareFloat, ok2 := compareFloat[op]
+	if !ok1 || !ok2 {
+		return nil, fmt.Errorf("unknown int comparator: %s", op)
+	}
+	// many different zeek data types can be compared with integers
+	return func(val zng.Value) bool {
+		zv := val.Bytes
+		switch val.Type.ID() {
+		case zng.IdInt8, zng.IdInt16, zng.IdInt32, zng.IdInt64:
+			v, err := zng.DecodeInt(zv)
+			if err == nil {
+				return CompareInt(v, pattern)
+			}
+		case zng.IdUint8, zng.IdUint16, zng.IdUint32, zng.IdUint64:
+			v, err := zng.DecodeUint(zv)
+			if err == nil && v <= math.MaxInt64 {
+				return CompareInt(int64(v), pattern)
+			}
+		case zng.IdFloat64:
+			v, err := zng.DecodeFloat64(zv)
+			if err == nil {
+				return CompareFloat(v, float64(pattern))
+			}
+		case zng.IdTime:
+			ts, err := zng.DecodeTime(zv)
+			if err == nil {
+				return CompareInt(int64(ts), pattern)
+			}
+		case zng.IdDuration:
+			v, err := zng.DecodeInt(zv)
+			if err == nil {
+				return CompareInt(int64(v), pattern)
+			}
```

Comment:
Maybe fold these two cases in with the signed integer cases?

Reply:
I did that first and the resulting bug is that int64 is scaled by 1_000_000 for time and dur comparisons as it presumes s not ns as the unit.